packages/cliqz-home-react/src/News.js

```diff
@@ -1,127 +1,51 @@
-    	});
+class News extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pageOfItems: [],
+      opacity: 0
     }
-    articles.forEach((el) => el.classList.add('opaque'));
-    clearTimeout(this._animationTimeout);
-
-    this._animationTimeout = setTimeout(() => {
-	    const start = this.state._currentPage * 3;
-	    const end = this.state._currentPage * 3 + 3;
-	    articles.forEach((el, i) => {
-        if (i >= start && i < end) {
-            el.style.display = 'block';
-            el.classList.remove('opaque');
-        } else {
-            el.style.display = 'none';
-        }
-	    });
-	    $$('a.button', app).forEach((e, i) => {
-	       e.classList[i === this.state._currentPage ? 'add' : 'remove']('active');
-	    });
-		}, 300);
-
-	}
-
-	_startPageSwitch() {
-		clearInterval(this._switchPageTimeout);
-		this._switchPageTimeout = setInterval(this._nextPage.bind(this), 5e3);
-	}
-
-	_nextPage() {
-		const currentPage = this.state._currentPage++;
-		this.setState({
-			_currentPage: currentPage
-		}, () => {
-			this._updatePage();
-		});
-	}
-
-	buttonClick(event) {
-		const el = event.target;
-		if(!el.classList.contains('button') || el.classList.contains('active')) {
-			return;
-		}
-
-		const currentPage = +el.dataset.index
-		this.setState({
-			_currentPage: currentPage
-		}, () => {
-			this._updatePage();
-			this._startPageSwitch();
-		});
-
-	}
-
-	render() {
-		return (
-			<div className="cliqz-news">
-
-				{
-					this.pages.map((page, i) => {
-						return <a href="#" key={i} className={"button " + (page.active ? 'active' : '')} onClick={this.buttonClick} data-index={i}></a>
-					})
-				}
-
-				<div className="acordion" style={{height: "141px"}} id="news">
-					{
-						this.props.news.data.map(function(article, i) {
-							return <a href={article.url} 
-												key={i} 
-												className="article" 
-												style={{height: '129px'}}
-												ref={(elem) => { this.articleElem = elem}}>
-								<div className="side-front">
-									<Logo logo={article.logo} />
-									<span className="source-name">
-										{article.displayUrl}
-									</span>
-									<p className="title">
-										{article.title}
-									</p>
-								</div>
 
-								<div className="side-back">
-									<p className="abstract">
-										{article.description}
-									</p>
-								</div>
-							</a>
-						}.bind(this))
-					}
-				</div>
-			</div>
-		);
-	}
+    this.onChangePage = this.onChangePage.bind(this);
+    this.updateOpacity = this.updateOpacity.bind(this);
+  }
+
+  onChangePage(pageOfItems) {
+    return Promise.resolve(this.setState({ pageOfItems: pageOfItems }));
+  }
+
+  updateOpacity(opacity) {
+    return Promise.resolve(this.setState({ opacity: opacity}))
+  }
+
+  render() {
+    return (
+      <div className="cliqz-news">
+
+        <Pagination items={this.props.news.data}
+                    onChangePage={this.onChangePage}
```

Comment:
we should discuss indentation strategy for JSX. Kinda like this one. Wonder what airbnb linter preset will tell about it.